docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install

```diff
@@ -1026,3 +1028,113 @@ function su_humsci_profile_update_9714() {
+      ->accessCheck(FALSE)
+      ->execute();
+    $sandbox['total'] = count($sandbox['ids']);
+  }
+  $paragraph_ids = array_splice($sandbox['ids'], 0, 50);
+
+  foreach ($paragraph_storage->loadMultiple($paragraph_ids) as $paragraph) {
+    $paragraph->delete();
+  }
+
+  $sandbox['#finished'] = count($sandbox['ids']) ? 1 - count($sandbox['ids']) / $sandbox['total'] : 1;
+}
+
+/**
+ * Delete hs_webform paragraph type and remove webform config.
+ */
+function su_humsci_profile_update_9716() {
+  $paragraph_machine_name = 'hs_webform';
+  // Delete the field and field storage on the paragraph.
+  $fields = \Drupal::entityTypeManager()
+    ->getStorage('field_config')
+    ->loadByProperties(['bundle' => $paragraph_machine_name]);
+
+  if ($fields) {
+    foreach ($fields as $field) {
+      /** @var Drupal\field\Entity $field */
+
+      // Delete the field configuration (instance) from the paragraph type.
+      $field->delete();
+    }
+  }
+
+  // Delete the field_hs_webform field storage.
+  $field_storage = FieldStorageConfig::load('field_hs_webform');
+  if ($field_storage) {
+    $field_storage->delete();
+  }
+
+  // Delete the paragraph itself.
+  $paragraph_type = ParagraphsType::load($paragraph_machine_name);
+  if ($paragraph_type) {
+    $paragraph_type->delete();
+  }
+
+  // Remove hs_webform from config ignored configs.
+  $config_storage = \Drupal::service('config.storage.sync');
+
+  $configs_to_change = [
+    'node.hs_basic_page.field_hs_page_components' => 'node.field_hs_page_components',
+    'paragraph.hs_collection.field_hs_collection_items' => 'paragraph.field_hs_collection_items',
+    'node.hs_private_page.field_hs_priv_page_components' => 'node.field_hs_priv_page_components',
+  ];
+
+  /** @var \Drupal\Core\Entity\EntityStorageInterface $field_config_storage */
+  $field_config_storage = \Drupal::service('entity_type.manager')->getStorage('field_config');
+
+  /** @var \Drupal\Core\Entity\EntityStorageInterface $fieldConfigStorage */
+  $field_storage = \Drupal::service('entity_type.manager')->getStorage('field_storage_config');
+
+  foreach ($configs_to_change as $field_config => $field_storage_config) {
+
+    // If storage does not yet exist, create it.
+    if (empty($field_storage->load($field_storage_config))) {
+      $field_storage->createFromStorageRecord($config_storage->read('field.storage.' . $field_storage_config))->save();
+    }
+
+    // Update the field instances from filesystem.
+    $config = $field_config_storage->load($field_config);
+    if (empty($config)) {
+      $field_config_storage->create($config_storage->read('field.field.' . $field_config))->save();
+    }
+  }
+
+  // Remove webform existing config - needs to be done before uninstall.
+  $query = \Drupal::database()
+    ->select('config', 'c')
+    ->fields('c', ['name'])
+    ->condition('name', '%' . Database::getConnection()->escapeLike('webform') . '%', 'LIKE')
+    ->execute();
+  $config_names = $query->fetchCol();
```

Comment:
Use `\Drupal::configFactory()->listAll('webform.')` instead of db query.

Reply:
I checked, and the the list of config items is a bit different when using `\Drupal::configFactory()->listAll('webform.')`, because we only get items that start with `webform`, while the db query includes items that include `webform` anywhere in the name:

- `editor.editor.webform_default`
- `filter.format.webform_default`
- `system.action.webform_archive_action`
- `system.action.webform_close_action`
- `system.action.webform_delete_action`
- `system.action.webform_open_action`
- `system.action.webform_submission_delete_action`
- `system.action.webform_submission_make_lock_action`
- `system.action.webform_submission_make_sticky_action`
- `system.action.webform_submission_make_unlock_action`
- `system.action.webform_submission_make_unsticky_action`
- `system.action.webform_unarchive_action`
- `ultimate_cron.job.webform_cron`

Reply:
Ah I see. Maybe just do `->listAll()` and then you can `array_filter` the config names that include webform. I'm just not liking a db query directly like this when there are other ways.

Reply:
@pookmish Yes, that makes sense. It's fixed now, thanks.